libsolidity/analysis/TypeChecker.cpp

```diff
@@ -807,12 +808,21 @@ bool TypeChecker::visit(TupleExpression const& _tuple)
 			{
 				components[i]->accept(*this);
 				types.push_back(type(*components[i]));
+				if (i == 0 && isArray)
+					internalArrayType = types[i]->mobileType();
+				else if (isArray)
+					internalArrayType = Type::commonType(internalArrayType, types[i]->mobileType());
+					
+				if (internalArrayType == nullptr && isArray) 
```

Comment:
Please use `!internalArrayType` instead of `internalArrayType == nullptr` (because it is a smart pointer).
Furhermore, please delay the fatal type error to after the loop (we want to do as much type checking as possible before reporting a fatal error that will stop the checking). Of course, if you do this, you have to check for internalArrayTYpe being empty before you call `Type::commonType`.
Oh and you might also check that `types[i]->mobileType()` is not empty (perhaps assert).